### ISSUE TITLE
Error Occurs When Trying to Retrieve CloudFormation Stack Output Using "${cf" with Nested "${file"

### ISSUE THREAD

**Issue by Hiroki-Aoki** [serverless/serverless#12822]
When deploying with serverless.yml to retrieve information from an existing CloudFormation stack as shown below, the variables cannot be resolved and an error occurs.

> None of the variables returned a value and no default value was provided. Please check your variable definitions or provide a default value.

It worked correctly in version 3 but causes an error in version 4.

I think this error is related to "${cf" and nested "${file". Will it be fixed?
 
Version used: Serverless Framework: 4.3.2

## Error Case

- serverless.yml

```yml
hoge: ${cf:${file(./fuga.yml):CloudFormationName}.attributeName}
```

- fuga.yml

```yml
cloudformationName: ${file(./foo.yml):AAAA}
```

- foo.yml

```yml
AAAA: abcd
```

## OK Case

In the following case, the error does not occur, and the variables can be resolved.

### OK Case 1

- serverless.yml

```yml
HOGE: ${file(./fuga.yml):CloudFormationName}
```

- fuga.yml

```yml
cloudformationName: ${file(./foo.yml):AAAA}
```

- foo.yml

```yml
AAAA: abcd
```

### OK Case 2

- serverless.yml

```
FUGA: ${cf:${file(./fuga.yml):CloudFormationName}.attributeName}
```

- fuga.yml

```yml
cloudformationName: abcd
```

**Comment by czubocha** [comment]
Hey @Hiroki-Aoki, this has been addressed in version 4.4.3, which is now live. Feel free to try it out and let us know if you run into any issues.

**Comment by Hiroki-Aoki** [comment]
@czubocha Thank you. I'm looking forward to the release of 4.4.3.
